mod-source-record-manager-server/src/main/java/org/folio/services/ParsedRecordServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package org.folio.services;
+
+import org.folio.HttpStatus;
+import org.folio.dataimport.util.OkapiConnectionParams;
+import org.folio.dataimport.util.Try;
+import org.folio.rest.client.SourceStorageClient;
+import org.folio.rest.jaxrs.model.ParsedRecordDto;
+import org.folio.rest.jaxrs.model.SourceRecord;
+import org.springframework.stereotype.Service;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+import javax.ws.rs.NotFoundException;
+
+import static org.folio.HttpStatus.HTTP_NOT_FOUND;
+import static org.folio.HttpStatus.HTTP_OK;
+
+@Service
+public class ParsedRecordServiceImpl implements ParsedRecordService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ParsedRecordServiceImpl.class);
+
+  @Override
+  public Future<ParsedRecordDto> getRecordByInstanceId(String instanceId, OkapiConnectionParams params) {
+    Promise<ParsedRecordDto> promise = Promise.promise();
+    SourceStorageClient client = new SourceStorageClient(params.getOkapiUrl(), params.getTenantId(), params.getToken());
+    try {
+      client.getSourceStorageSourceRecordsById(instanceId, "INSTANCE", response -> {
+        if (HTTP_OK.toInt() == response.statusCode()) {
+          response.bodyHandler(body -> promise.handle(Try.itGet(() -> mapSourceRecordToParsedRecordDto(body))));
+        } else {
```

Comment:
I propose to respond 404 in case we get 404 from SRS. What do you think @KaterynaSenchenko

Reply:
was too lazy to do it, but OK, added

Reply:
Thank you)!